### PR TITLE
Fix squares sample failure when a single file passed

### PR DESCRIPTION
**Pull request by alexkalmuk**
Fix samples/cpp/squares.cpp:

```
$ ../build/opencv_squares pic1.png 

A program using pyramid scaling, Canny, contours and contour simplification
to find squares in a list of images (pic1-6.png)
Returns sequence of squares detected on the image.
Call:
./../build/opencv_squares [file_name (optional)]
Using OpenCV version 4.5.2-dev

[ WARN:0] global /home/alexander/git/opencv/modules/core/src/utils/samples.cpp (59) findFile cv::samples::findFile('0') => ''
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.5.2-dev) /home/alexander/git/opencv/modules/core/src/utils/samples.cpp:62: error: (-2:Unspecified error) OpenCV samples: Can't find required data file: 0 in function 'findFile'

Aborted (core dumped)

```

It tries to find file "0" and fails.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/cpp/squares.cpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ int main(int argc, char** argv)
     if( argc > 1)
     {
      names[0] =  argv[1];
-     names[1] =  "0";
+     names[1] =  0;
```
Comment (alalek) [inline samples/cpp/squares.cpp:133]: Could you please remove `static` from `names` declaration too (doesn't make sense):

```.diff
-static const char* names
+const char* names
```

**Review by alalek — COMMENTED** [review]
Thank you for contribution!

This patch should go into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first.
We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

Please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".

File: samples/cpp/squares.cpp (hunk 1 above)
Comment (alexkalmuk) [reply samples/cpp/squares.cpp:133]: Yes, of course. I fixed this and squashed it in a single commit.

**Comment by alexkalmuk** [comment]
@alalek Thanks for the tips! Done.

**Review by alalek — APPROVED** [review]
Well done :+1:

**Pull request merged**